swarm/api/client/client.go

```diff
@@ -665,9 +643,9 @@ func (c *Client) GetResourceMetadata(manifestAddressOrDomain string) (*mru.Reque
 		return nil, err
 	}
 
-	metadata, err := mru.DecodeUpdateRequest(body)
-	if err != nil {
+	var metadata mru.Request
+	if err := metadata.Unmarshal(body); err != nil {
```

Comment:
I don't see how `json.Unmarshal` applies here. `json.Unmarshal` is a generic unmarshaller and needs an interface passed by ref so it can use reflection to examine what type you actually passed in (slower). For what we care in our use case, the passed value is an object serialized as a byte array which just happens to have been serialized as JSON, but could have been serialized in any other way.

Reply:
Hey no it doesn't apply here but it does apply in the `Unmarshal` method on `metadata` - I just put the comment in the wrong place.

Reply:
I don't quite understand, can you highlight the specific code and point me to that interface in the documentation?

Reply:
See request.go line 251, and godoc `encoding/json` interface `Unmarshal`

Reply:
ok, I see. I have implemented the interface. Turns out it was just a method name change.